src/main/java/services/DBService.java

```diff
@@ -37,4 +42,14 @@ public User getUserDetails(int userID) throws ExecutionException, InterruptedExc
 
     }
 
+    public void addContact(User user, Integer contactID) throws ExecutionException, InterruptedException {
+        Firestore dbFireStore = FirestoreClient.getFirestore();
+        System.out.println(user.getContacts());
+        String docName = "id" + user.getUser_id();
+        System.out.println(docName);
+        user.getContacts().add(contactID);
+        System.out.println(user.getContacts());
```

Comment:
Add this to successfully add a a contact:
DocumentReference docRef = dbFireStore.collection("users").document(docName);
System.out.println(docRef);
ApiFuture<WriteResult> future = docRef.update("contacts", user.getContacts());
      WriteResult result = future.get();

Reply:
ok thanks for the advice!

Reply:
I think to add a contact, the correct code would be:

**_DocumentReference userRef = dbFireStore.collection("users").document(docName);
DocumentReference contactRef = dbFireStore.collection("users").document("id" + contactID);
try {
    DocumentSnapshot userDoc = userRef.get().get();
    Map<String, Object> userFields = userDoc.getData();
    ArrayList<DocumentReference> currContacts = (ArrayList<DocumentReference>) userFields.get("contacts");
    currContacts.add(contactRef);
    ApiFuture future = userRef.set(userFields);
    WriteResult result = future.get();
    
} catch (ExecutionException | InterruptionException e) {
    throw new RuntimeException(e);
}_**

This is because a user's contacts field in the Firebase database isn't a list of User objects (because Firebase doesn't know what a User object is). So it actually stores a list of DocumentReference objects that refer to other users in the database. 
Also, we have a try-catch block because the userRef.get().get() method could possibly throw an ExecutionException or InterruptionException (which are subclasses of RuntimeException).

Hope this helps!